docs/ai-usage.md

```
@@ -215,3 +215,153 @@ React Flow 형태(`CustomEditorNode`)를 캐시에 넣지 않기로 한 이유
+- `useFolderPath`는 `{ id, name }[]` 스택만 소유하고 폴더 목록은 들고 있지 않다. 목록의 소유자는 query 하나뿐이다.
+- 브라우저에서 시나리오를 재현해 확인했다. 자식 폴더에 들어간 직후 "상위 폴더로"를 누르면 10ms 뒤 부모 목록이 **스켈레톤 없이** 복원되고(`skeletons: 0`), 1.5초 뒤에도 자식 폴더의 응답이 화면을 덮어쓰지 않았다.
+
+---
+
+## 사례 5. 모달 상태를 zustand에 둘 것인가 — 리렌더 범위는 상태의 위치가 정한다
+
+**대상 파일**: `src/shared/ui/modal/Modal.tsx`, `app/(main)/recent/page.tsx`
+
+### 상황
+
+#61 계획의 4단계에서 AI가 최신순 화면에 팝업 열림 상태를 `useState`로 두는 안을 제시했다.
```

markdownlint-cli2 (0.23.2)

[warning] No space after hash on atx style heading

(MD018, no-missing-space-atx)